### ISSUE TITLE
Add status bar to core

### ISSUE THREAD

**Issue by saulshanabrook** [jupyterlab/jupyterlab#5352]
It would be great to merge the status bar into core! https://github.com/jupyterlab/jupyterlab-statusbar

**Comment by saulshanabrook** [comment]
@ian-r-rose I know there has been some prior work on how to do this sort of merge, but wanted to throw the idea of [using git subtree to add the other repo to this one](https://medium.com/@porteneuve/mastering-git-subtrees-943d29a798ec#defe) then adding commits on top of that to do any cleaup that needs to happen.

**Comment by ian-r-rose** [comment]
Yep, that was my plan. I have done that before, but every time I do it I have to re-learn it :smile:

**Comment by saulshanabrook** [comment]
Yep that makes sense. I am interested in seeing how it is done, so if it works could you ping me when you are going to do it and maybe we could do a screen share?

**Comment by ian-r-rose** [comment]
Sure thing. We can try to stumble through it together.

**Comment by ellisonbg** [comment]
Is the plan to move the individual status bar items to the extensions that deal with that stuff. For example, would notebook related things get moved to the notebook extension etc? That way the core status bar extension could just be the core APIs that other extensions use? It may not be that simple, but wondering if that makes sense.

**Comment by ian-r-rose** [comment]
It think that makes sense. I think I'll have a better idea of it once we get the ball rolling.
